src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,79 @@
+using Flux
+
+function basicblock(inplanes::Int, planes::Int, stride::Int, dilation::Int)
+  basic =  Chain(
+      Conv((3, 3), inplanes=>planes, stride=stride, pad=(1,1), dilation=dilation),
+      BatchNorm(planes, λ = relu),
+      Conv((3, 3), planes=>planes, stride=stride, pad=(1,1), dilation=dilation),
+      BatchNorm(planes, λ=relu)
+    );
+  return basic
+end
+
+function bottleneck(inplanes::Int, planes::Int)
+  base_width = 64;
+  groups = 1;
+  width = Int(planes * (base_width / 64.)) * groups;
+  residual =  Chain(Conv((1, 1), inplanes=>width, stride=(2,2)),
+          BatchNorm(width, λ=relu),
+          Conv((3, 3), width=>width, stride=(2,2), pad=(1,1), dilation=dilation),
+          BatchNorm(width, λ=relu),
+          Conv((1, 1), width=>planes * expansion, stride=(2,2)),
+          BatchNorm(planes * expansion, λ=relu));
+  return residual
+end 
```

Comment:
I am referencing the table below (Table 1 from the paper). I am not sure what the logic with `width` is supposed to do here, but I think it can be simplified greatly. As you can see from the table (columns for ResNet-50+), each bottleneck block contains three convolutions, where the first two have `inplanes => inplanes` and the last is `inplanes` to `planes`. Therefore, I think we can simply use the following for the bottleneck (I replaced `planes` with `outplanes` for clarity):
```julia
bottleneck(inplanes, outplanes, downsample = false) = downsample ?
  Chain(Conv((1, 1), inplanes => inplanes, stride = 1),
        BatchNorm(inplanes, λ = relu),
        Conv((3, 3), inplanes => inplanes, stride = 1, pad = 1),
        BatchNorm(inplanes, λ = relu),
        Conv((1, 1), inplanes => outplanes, stride = 2),
        BatchNorm(outplanes, λ = relu)) :
  Chain(Conv((1, 1), inplanes => inplanes, stride = 1),
        BatchNorm(inplanes, λ = relu),
        Conv((3, 3), inplanes => inplanes, stride = 1, pad = 1),
        BatchNorm(inplanes, λ = relu),
        Conv((1, 1), inplanes => outplanes, stride = 1),
        BatchNorm(outplanes, λ = relu))
```

<img width="686" alt="Screen Shot 2020-06-29 at 10 45 22 AM" src="https://user-images.githubusercontent.com/4811350/86020253-4d179400-b9ed-11ea-9a2c-652cee3720b3.png">